api/v1alpha1/openapi.yaml

```diff
@@ -829,6 +835,9 @@ components:
     DeviceSpec:
       type: object
       properties:
+        fleetName:
```

Comment:
Making this part of the device spec implies the user explicitly specifies the fleet, which implies that fleet must exist and it inverts the logic of fleet label-selecting devices. Unless we want to abandon label-selection, I'd suggest moving this out of spec, either into a metadata field that's set by the fleet controller (maybe "ownerRef" or an annotation).

Reply:
How does the LabelSelector work - what if a device's labels match two fleet specs?  Does it make sense to reimage a device based on a label?

Reply:
I think label selection is a bad idea, because we can have a device being selected by multiple fleets.

I believe it's a safer route to do explicit assignments.

Reply:
I thought of using ownerRef's ... it's an option, I think it's better than label selections. Still at the database level we need to extract the fleet name, and have an index of that... otherwise there is no easy way to pull al devices from a fleet in a query.

Reply:
I don't think it's an owner - it's a device property

Reply:
Yeah, I agree to the problem of devices being selected by multiple fleets. That can also happen in K8s and there it's seen as "user problem". Then again, mistakes are less costly to fix in K8s.

In principle, I'm ok with inverting, but remember we don't want users to have to manipulate device resources but to work at the fleet resource level instead. Think about what you'd store in Git to fully describe the system. If we need to start storing/updating thousands of device manifests in Git as a result that'd not be a good design.

Also, the device's spec is a copy / rendering of the fleet's device template, so I don't think the fleet reference should be there.